src/main/java/edu/agh/wfiis/designpatterns/patternsUsageExample/DesignPatternsManager.java

```diff
@@ -1,47 +1,53 @@
 package edu.agh.wfiis.designpatterns.patternsUsageExample;
 
+import java.util.*;
+import java.io.*;
 
-import java.io.File;
-
-public class DesignPatternsManager {
-
-    public DesignPattern create(PatternName patternName) {
-        switch (patternName) {
-            case BUILDER: {
-                DesignPattern pattern = new DesignPattern("1","builder","builder pattern",null, true, false, false);
-                return pattern;
-            }
-            case DECORATOR: {
-                DesignPattern pattern = new DesignPattern("2","decorator","decorator pattern",null, true, false, false);
-                return pattern;
-            }
-            case STRATEGY: {
-                DesignPattern pattern = new DesignPattern("3","strategy","strategy pattern",null, false, false, true);
-                return pattern;
-            }
-            default: {
-                DesignPattern pattern = new UndefinedPattern();
-                return pattern;
-            }
+public class DesignPatternsManager implements Serializable {
+
+
+    private static volatile DesignPatternsManager instance;
+    private Map<String, DesignPattern> patterns = new HashMap<String, DesignPattern>();
```

Comment:
Nie trzeba podawać drugi raz typów budując mapę.
```suggestion
    private Map<String, DesignPattern> patterns = new HashMap<>();
```